core_v2/sources/v2_config.move

```diff
@@ -0,0 +1,388 @@
+    /// Alternatively, the deployer can be used to perform admin actions.
+    public fun signer_is_admin(sign: &signer): bool acquires Config {
+        signer::address_of(sign) == admin_address() || signer::address_of(sign) == @aptos_names_v2
+    }
+
+    public fun assert_signer_is_admin(sign: &signer) acquires Config {
+        assert!(signer_is_admin(sign), error::permission_denied(ENOT_AUTHORIZED));
+    }
+
+    //
+    // Setters
+    //
+
+    public entry fun set_is_enabled(sign: &signer, enabled: bool) acquires Config {
+        assert_signer_is_admin(sign);
+        borrow_global_mut<Config>(@aptos_names_v2).enabled = enabled
+    }
+
+    public entry fun set_fund_destination_address(sign: &signer, addr: address) acquires Config {
+        assert_signer_is_admin(sign);
+        aptos_account::assert_account_is_registered_for_apt(addr);
+        borrow_global_mut<Config>(@aptos_names_v2).fund_destination_address = addr
+    }
+
+    public entry fun set_admin_address(sign: &signer, addr: address) acquires Config {
+        assert_signer_is_admin(sign);
+        assert!(account::exists_at(addr), error::invalid_argument(EINVALID_VALUE));
+        borrow_global_mut<Config>(@aptos_names_v2).admin_address = addr
+    }
+
+    public entry fun set_max_number_of_seconds_registered(sign: &signer, max_seconds_registered: u64) acquires Config {
+        assert_signer_is_admin(sign);
+        assert!(max_seconds_registered > 0, error::invalid_argument(EINVALID_VALUE));
+        borrow_global_mut<Config>(@aptos_names_v2).max_number_of_seconds_registered = max_seconds_registered
+    }
+
+    public entry fun set_max_domain_length(sign: &signer, domain_length: u64) acquires Config {
+        assert_signer_is_admin(sign);
+        assert!(domain_length > 0, error::invalid_argument(EINVALID_VALUE));
+        borrow_global_mut<Config>(@aptos_names_v2).max_domain_length = domain_length
+    }
+
+    public entry fun set_min_domain_length(sign: &signer, domain_length: u64) acquires Config {
+        assert_signer_is_admin(sign);
+        assert!(domain_length > 0, error::invalid_argument(EINVALID_VALUE));
+        borrow_global_mut<Config>(@aptos_names_v2).min_domain_length = domain_length
+    }
+
+    public entry fun set_tokendata_description(sign: &signer, description: String) acquires Config {
+        assert_signer_is_admin(sign);
+        borrow_global_mut<Config>(@aptos_names_v2).tokendata_description = description
+    }
+
+    public entry fun set_tokendata_url_prefix(sign: &signer, url_prefix: String) acquires Config {
+        assert_signer_is_admin(sign);
+        borrow_global_mut<Config>(@aptos_names_v2).tokendata_url_prefix = url_prefix
+    }
+
+    public entry fun set_subdomain_price(sign: &signer, price: u64) acquires Config {
+        assert_signer_is_admin(sign);
+        borrow_global_mut<Config>(@aptos_names_v2).subdomain_price = price
```

Comment:
in the future you can further reduce this with a inline

```
inline fun authorized_borrow_mut(sign): &mut Config acquires Config {
  assert_signer_is_admin(sign);
  borrow_global_mut<Config>(@aptos_names_v2)
}
```

then you can reduce each of these by 50%